keras/engine/topology.py

```diff
@@ -2552,6 +2565,54 @@ def load_weights_from_hdf5_group(self, f):
                 weight_value_tuples += zip(symbolic_weights, weight_values)
             K.batch_set_value(weight_value_tuples)
 
+    def load_weights_from_hdf5_group_by_name(self, f):
+        '''Weight loading is based on layer names
+        (not by sequential order).
+        Layers that have no matching name are skipped.
+        '''
+        if hasattr(self, 'flattened_layers'):
+            # support for legacy Sequential/Merge behavior
+            flattened_layers = self.flattened_layers
+        else:
+            flattened_layers = self.layers
+
+        if 'nb_layers' in f.attrs:
+                raise Exception('You are trying to load a weight file ' +
+                                'by name with an unsupported legacy' +
+                                'format.')
```

Comment:
"The weight file you are trying to load is in a legacy format that does not support name-based weight loading".